pygsti/objects/operation.py

```diff
@@ -2742,7 +2847,7 @@ def copy(self, parent=None, memo=None):
             A copy of this object.
         """
         if memo is not None and id(self) in memo: return memo[id(self)]
-        copyOfMe = StochasticNoiseOp(self.dim, self.basis, self._evotype, self._params_to_rates(self.to_vector()))
+        copyOfMe = StochasticNoiseOp(self.basis.size, self.basis, self._evotype, self._params_to_rates(self.to_vector()))
```

Comment:
Does self.dim not work?

Reply:
dim was being used as the number of superoperators in the expansion. This change was reverted and I made sure to use basis.size more careful in StochasticNoiseOp and DepolarizeOp.